### PR TITLE
[java][bidi] Add LogInspector

### PR DESCRIPTION
**Pull request by pujagani**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description

Add LogInspector contains the w3c Bidi spec related APIs to listen to log events.

### Motivation and Context

Refer to https://docs.google.com/document/d/1dCd8Y2PYaR5mOGSmNTwllEHNmFqegfoGkP-TCKvPzSU/edit?usp=sharing about the details. 

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [X] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: java/src/org/openqa/selenium/bidi/LogInspector.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  private final List<Consumer<GenericLogEntry>> genericLogListeners = new LinkedList<>();
+  private final Set<String> browsingContextIds;
+
+  private final BiDi bidi;
+
+  public LogInspector(WebDriver driver) {
+    this(new HashSet<>(), driver);
+  }
+
+  public LogInspector(String browsingContextId, WebDriver driver) {
```
Comment (krmahadevan) [inline java/src/org/openqa/selenium/bidi/LogInspector.java:51]: Maybe we could just do `this(Collections.singleton(browsingContextId), driver);`

File: java/src/org/openqa/selenium/bidi/LogInspector.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
+package org.openqa.selenium.bidi;
```
Comment (krmahadevan) [inline java/src/org/openqa/selenium/bidi/LogInspector.java:18]: Also I think the license header needs to be applied to both the classes (I am not sure if it gets auto applied in some way. If yes, please let me know and I will use that same mechanism as well)

Comment (titusfortner) [reply java/src/org/openqa/selenium/bidi/LogInspector.java:18]: we periodically run this to make sure everything is included: `./go copyright:update`

File: java/src/org/openqa/selenium/bidi/LogInspector.java (hunk 1 above)
Comment (pujagani) [reply java/src/org/openqa/selenium/bidi/LogInspector.java:51]: We can call the constructor but it does not allow me to do null checks in that case. Which needs to happen before we call the constructor but when calling using `this(Collections.singleton(browsingContextId), driver);` it has to be the first statement.

Comment (krmahadevan) [reply java/src/org/openqa/selenium/bidi/LogInspector.java:51]: @pujagani - The null check for the contents of the collection would already be done by `Collections.singleton()` so we don't need to be doing it again. If we need it to be done explicitly we can create a private static method 

```java
private static Set<String> asSet(String text) {
    Require.nonNull("Browsing context id", text);
    return Collections.singleton(text);
}
```

and now within the constructor you can do 

```java
this(asSet(browsingContext), driver);
```

What do you think ?

Comment (krmahadevan) [reply java/src/org/openqa/selenium/bidi/LogInspector.java:51]: Ok. So `Collections.singleton()` doesn't do a null check.My bad. But my suggestion to do it via a static method should still do the trick and allow us to chain constructor calls.

Comment (pujagani) [reply java/src/org/openqa/selenium/bidi/LogInspector.java:51]: Thank you for your suggestion. The `Require.nonNull` does that already, it returns a value. I somehow missed that. Now we are able to chain constructors.

**Review by diemol — APPROVED** [review]
Thank you, @pujagani!

**Pull request merged**